enhancements/windows-containers/windows-node-egress-proxy.md

```diff
@@ -167,27 +168,33 @@ OLM is a subscriber to these `Proxy` settings -- it forwards the settings to the
 container will automatically get the required `NO_PROXY`, `HTTP_PROXY`, and `HTTPS_PROXY` environment variables on startup.
 In fact, OLM will update and restart the operator pod with proper environment variables if the `Proxy` resource changes.
 
-On the Linux side this is achieved through `MachineConfigs` -- when global proxy settings are changed, the ignition
-config is re-rendered by MCO (for whom proxy info is injected by CVO) to include a systemd unit
-`EnvironmentFile=/etc/mco/proxy.env`, with proxy vars set in this file.
-
-WMCO will retrieve the proxy variables by watching the `rendered-worker` `MachineConfig` for changes and parse info from
-the `proxy.env` file. This could be an expensive operation, but since we already parse ignition from the `MachineConfig`
-as part of node bootstrap, the technical cost and complexity is reduced.
-
-Then, once it has the required values, the operator will set the 3 proxy environment variables on the Windows instance
-during node configuration/reconciling through Powershell.
-```powershell
-[Environment]::SetEnvironmentVariable('<name>', '<value>', 'Machine')
+WMCO the store the 3 proxy variables and their values in the `windows-services` ConfigMap.
+This will be done by adding a `EnvironmentVars` key to the services ConfigMap spec.
+The new key will be a list of name-value pairs representing environment variables to be set on Windows instances:
+```json
+[
+  {
+    "name": "name of the environment variable",
+    "value": "value the variable should be set to on the instance"
+  }
+]
 ```
-Note that we will set these variables at the `Machine` scope, the highest level, to ensure all WMCO-managed services
-(kubelet, containerd, etc.) will inherit them. Since Windows services read environment vars from the system registry,
-and the registry does not update until a shutdown/startup cycle, this will require rebooting the instance.
+
+WICD, which uses the ConfigMap data to configure Windows services, will now have the added responsibility to ensure the
+proper setting of these `EnvironmentVars` variables on the instance. 
+1. WICD will check if any env vars changed by comparing each variable's value on the node to the expected value in the
+   ConfigMap. If there is a discrepancy:
+  - WICD will update the env vars if needed, e.g.
+    ```powershell
+    $Env:HTTP_PROXY = "http://<username>:<pswd>@<ip>:<port>"
+    $Env:NO_PROXY = "123.example.com,10.88.0.0/16"
+    ```
```

Comment:
> This solution isn't specific to a service, what if a service and another service need different values for same variable?

@sebsoto There is already a solution to this. The `EnvironmentVars` will be the default, but if a specific service wants to override them, it can set it's own by providing a powershell prescript `Set-ItemProperty ...`

> add support to WICD to execute PowerShell scripts independent of a service

I don't think we need to introduce this. The sys/windows package does have a call to set env vars so it should be simple. WICD can already run powershell directly on the host anyway.

Reply:
How does this solve the reboot issue if you are not applying the environment variables to the service's themselves, and you are instead setting them for the powershell environment?

Reply:
Also, a powershell pre-script isn't going to work if its executed before the service is created.

Reply:
>WICD can already run powershell directly on the host anyway.

Only in the context of a service. At some point we might need to introduce running of PowerShell scripts that has nothing to do with a service and was wondering if we need to do that in this case.